src/value/value.rs

```diff
@@ -172,6 +172,26 @@ impl Value {
     }
 }
 
+impl PartialOrd for Value {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        if std::mem::discriminant(self) != std::mem::discriminant(other) {
+            return None;
+        }
+        match (self, other) {
+            (Value::Bytes(a), Value::Bytes(b)) => a.partial_cmp(b),
+            (Value::Regex(a), Value::Regex(b)) => a.partial_cmp(b),
+            (Value::Integer(a), Value::Integer(b)) => a.partial_cmp(b),
+            (Value::Float(a), Value::Float(b)) => a.partial_cmp(b),
+            (Value::Boolean(a), Value::Boolean(b)) => a.partial_cmp(b),
+            (Value::Timestamp(a), Value::Timestamp(b)) => a.partial_cmp(b),
+            (Value::Object(a), Value::Object(b)) => a.partial_cmp(b),
+            (Value::Array(a), Value::Array(b)) => a.partial_cmp(b),
+            (Value::Null, Value::Null) => Some(Ordering::Equal),
```

Comment:
The semantics here are quite tricky, for example in the SQL space `null =/= null`. If we are compared `null` variable in Java those would be true. I am tempted to say we should just remove this case and let it go to `None`.

Reply:
Left it as `None`, if someone has a better case for `null == null` we can change it later.